subsys/net/lib/app/init.c

```diff
@@ -238,6 +238,11 @@ int net_app_init(const char *app_info, u32_t flags, s32_t timeout)
 		NET_INFO("%s", app_info);
 	}
 
+	if (!iface) {
```

Comment:
This is why https://jira.zephyrproject.org/browse/ZEP-2105 is still open, waiting for the general fix of this issue. In the meantime, would be nice to get this into 1.9, because without it, switching from adhoc network config, but with no-interface check, to net_app_setting would lead to a crash.